controller/scheduler/scheduler.go

```diff
@@ -722,51 +734,64 @@ func (s *Scheduler) changeFormation(ef *ct.ExpandedFormation) (*Formation, error
 		}
 	}
 
-	f := s.formations.Get(ef.App.ID, ef.Release.ID)
+	f = s.formations.Get(ef.App.ID, ef.Release.ID)
 	if f == nil {
 		log.Info("adding new formation", "processes", ef.Processes)
-		f = s.formations.Add(NewFormation(ef))
+		f = s.formations.Add(NewFormation(ef, s.HandleRectify))
 	} else {
-		if !f.GetProcesses().Equals(ef.Processes) {
+		if f.GetProcesses().Equals(ef.Processes) {
+			return f, nil
+		} else {
 			log.Info("updating processes of existing formation", "processes", ef.Processes)
 			f.Processes = ef.Processes
 		}
 	}
+	s.triggerRectify(f.key())
 	return f, nil
 }
 
+func (s *Scheduler) triggerRectify(key utils.FormationKey) {
+	logger.Info("triggering rectify", "key", key)
+	s.formations.TriggerRectify(key)
```

Comment:
Ok so I am not really a fan of the reflection and a channel per formation just to avoid the chance of a loop iteration triggering the same rectify multiple times, as the duplicates will effectively be no-ops anyway right? The loop should be fast enough that the chance of having duplicates is low. WDYT?